test/e2e/cypress/integration/clusters_overview.js

```diff
@@ -44,5 +44,226 @@ context('Clusters Overview', () => {
         cy.get(`a:contains(${clusterID})`).should('be.visible');
       });
     });
+
+    describe('Health status for each cluster is correct', () => {
+      const healthyClusterScenario = {
+        clusterName: 'hana_cluster_2',
+        checks: ['156F64'],
+        hostIds: [
+          '99cf8a3a-48d6-57a4-b302-6e4482227ab6',
+          'e0c182db-32ff-55c6-a9eb-2b82dd21bc8b',
+        ],
+        result: 'passing',
+        executionId: '8d840ba1-828f-4cb8-bce0-6e34224fe4f8',
+      };
+
+      const unhealthyClusterScenario = {
+        clusterName: 'hana_cluster_3',
+        checks: ['156F64'],
+        hostIds: [
+          '9cd46919-5f19-59aa-993e-cf3736c71053',
+          'b767b3e9-e802-587e-a442-541d093b86b9',
+        ],
+        result: 'critical',
+        executionId: '8d840ba1-828f-4cb8-bce0-6e34224fe4f9',
+      };
+
+      before(() => {
+        cy.task('selectChecks', {
+          clusterId: clusterIdByName(healthyClusterScenario.clusterName),
+          checkIds: healthyClusterScenario.checks,
+        });
+        cy.task('mockStartChecksExecution', {
```

Comment:
for an execution to start on the runner there's the need to have first requested for that execution to start.

I am not completely sure the issue you're referring to is an actual bug, I have the impression we're missing something in the flow.
is the command RequestChecksExecution  being dispatched to the aggregate? And the following ChecksExecutionRequested  event emitted?

Reply:
No, I was actually avoiding it as I understood that this might create the random mocked results, which don't allow the test to fail or succeed in a predictable way.

Is there any better way to do this?

Reply:
I understand, but you _have to_ go through that command/event pair, because it changes state of the Aggregate.

You might need to look into `Trento.Integration.Checks.MockRunner` which should allow you to set some `expected_results`.

I don't know whether that's enough or some extra pieces are needed to instruct during e2e tests.

The thing I am pretty sure is that that piece of the flow is missing, and some business logic has not been applied.

Reply:
> You might need to look into Trento.Integration.Checks.MockRunner which should allow you to set some expected_results.

Thanks! I'll take a look at `Trento.Integration.Checks.MockRunner`

>The thing I am pretty sure is that that piece of the flow is missing, and some business logic has not been applied.

Yes, agreed 100%. As I later clarified in slack, I don't think this exposes a bug per se as it's an edge case that a runner would reach the callback endpoint without a prior request. It would be more of an improvement: handling those scenarios correctly by e.g. discarding the event.

Reply:
> handling those scenarios correctly by e.g. discarding the event.
>
What does this mean? :thinking:

Reply:
Let's try to create an additional e2e environment so we can enable the actual runner, instead of the mocked one, as it doesn't allow us doing real integration tests

Reply:
> > handling those scenarios correctly by e.g. discarding the event.
> 
> What does this mean? thinking

This is about the original issue I was exposing:
the `Elixir.Trento.Domain.Events.ChecksExecutionCompleted` gets stuck endlessly as the projection fails with `(Ecto.StaleEntryError) attempted to update a stale struct:`. 

> Let's try to create an additional e2e environment so we can enable the actual runner, instead of the mocked one, as it doesn't allow us doing real integration tests

Yes, I do actually think this is the right path :+1: